modules/assim.sequential/R/Multi_Site_Constructors.R

```diff
@@ -0,0 +1,258 @@
+##' @title Contruct.Pf
+##' @name  Contruct.Pf
+##' @author Hamze Dokoohaki
+##' 
+##' @param site.ids a vector name of site ids.  
+##' @param var.names vector names of state variable names.
+##' @param X a matrix of state variables. In this matrix rows represent ensembles, while columns show the variables for different sites.
+##' @param localization.FUN This is the function that performs the localization of the Pf matrix and it returns a localized matrix with the same dimensions.
+##' @description The argument X needs to have an attribute pointing the state variables to their corresponding site. This attribute needs to be called `Site`.
+##' At the moment, the cov between state variables at blocks defining the cov between two sites are assumed zero.
+##' @return It returns the var-cov matrix of state variables at multiple sites.
+##' @export 
+
+
+Contruct.Pf <- function(site.ids, var.names, X, localization.FUN=NULL, t=1, blocked.dis=NULL, ...) {
+  #setup
+  nsite <- length(site.ids)
+  nvariable <- length(var.names)
+  # I will make a big cov matrix and then I will populate it with the cov of each site
+  pf.matrix <-matrix(0,(nsite*nvariable),(nsite*nvariable))
```

Comment:
high level question: I don't understand why pf.matrix isn't just cov(X) on the full X. What am I missing in why you put so much work into calculating the matrix block-by-block

Reply:
This is the piece I was trying to explain today that, I do this do zero out the off-diagonal cov matrix between two sites.